---
title: "sclerotinia, iatab"
author: "Alfonso R. Reyes"
date: Version of `r format(Sys.time())`
output:
  bookdown::pdf_document2:
    keep_tex: yes
    latex_engine: pdflatex
    number_sections: no
    toc: yes
  bookdown::html_document2:
    number_sections: no
    theme: readable
    toc: yes
  bookdown::tufte_html2:
    number_sections: no
    toc: yes
  bookdown::word_document2: null
fontsize: 12pt
linestretch: 1.0
link-citations: yes
---


```{r setup, include=FALSE, error=TRUE, message=FALSE, warning=FALSE} 
knitr::opts_chunk$set(echo = FALSE, 
                      comment = NA, 
                      error = TRUE, 
                      warning = FALSE, 
                      message = FALSE, 
                      fig.align = 'center'
                      )
```



```{r}
library("tidyverse")
```


```{r}
gdt <- read_csv("genotype_diversity_table_region.csv", 
         na = "NA", # spaces read in as spaces
         col_types = cols(
         Pop = col_character(),
         N = col_integer(),
         MLG = col_integer(),
         eMLG = col_double(),
         SE = col_double(),
         H = col_double(),
         G = col_double(),
         lambda = col_double(),
         E.5 = col_double(),
         Hexp = col_double(),
         Ia = col_double(),
         p.Ia = col_double(),
         rbarD = col_double(),
         p.rD = col_double(),
         File = col_character()
      ))

gdtrd <- setNames(signif(gdt$p.rD, 3), gdt$Pop)
```



This indicated that there are few duplicated genotypes in WA (Table \@ref(tab:iatab)).
A more useful metric to compare populations, however, is $E_5$, which scales from 0 to 1, where 1 indicates all unique genotypes [@grunwald2003analysis]. 
Evaluating by $E_5$ shows that both MI and NE exhibit lower than average values, indicating that there are over-represented genotypes in the popualtions (table \@ref(tab:iatab)). 
When we look at Mexico, we observed that it had relatively high values of $E_5$ and genotypic diversity, but low richness, as measured by *eMLG*.
Moreover, Mexico had the lowest value for *h*, which is a measure of allelic diversity.
Nearly all populations showed evidence of linkage (Table \@ref(tab:iatab)), which serves as evidence for clonal reproduction or other forms of non-random mating. 
The only exceptions were CA (*P* = `r gdtrd[["CA"]]`) and Australia (*P* = `r gdtrd[["Australia"]]`). Both of these populations showed only moderate significance with $\bar{r}_d$ values of 0.03 and 0.12, respectively. 




```{r table_ia, echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
gdt %>%
  # mutate(Ia = case_when(
  #   !is.finite(p.Ia) ~ "-",
  #   p.Ia == 0.001 ~ paste0(round(Ia, 2), "*"),
  #   p.Ia <= 0.01  ~ paste0(round(Ia, 2), "-"),
  #   p.Ia <= 0.05  ~ paste0(round(Ia, 2), "~"),
  #   TRUE          ~ as.character(round(Ia, 2))
  # )) %>%
  mutate(rbarD = case_when(
    !is.finite(p.rD) ~ "-",
    p.rD == 0.001 ~ paste0(round(rbarD, 2), "*"),
  #   p.rD <= 0.01  ~ paste0(round(rbarD, 2), "-"),
  #   p.rD <= 0.05  ~ paste0(round(rbarD, 2), "~"),
    TRUE          ~ as.character(round(rbarD, 2))
  )) %>%
  mutate_if(is.numeric, round, digits = 2) %>%
  mutate(N = paste(N, MLG, sep = " (") %>% paste0(")") %>% formatC()) %>%
  mutate(eMLG = paste(eMLG, SE, sep = " (") %>%  paste0(")") %>% formatC()) %>%
  rename(`*h*` = Hexp) %>%
  rename(`$E_5$` = E.5) %>%
  # rename(`$I_A$` = Ia) %>%
  rename(`$\\lambda$` = lambda) %>%
  rename(`$\\bar{r}_d$` = rbarD) %>%
  rename(eMLH = eMLG) %>%
  select(-File, -MLG, -SE, -p.Ia, -p.rD, -Ia) %>% 
  huxtable::as_huxtable(add_colnames = TRUE) %>%
  huxtable::set_number_format(huxtable::everywhere, 4:5, 1) %>%
  huxtable::set_align(huxtable::everywhere, -1, "right") %>%
  huxtable::set_col_width(c(1.1, 0.9, 1.3, 0.5, 0.5, 1.1, 0.7, 0.6, 0.7)) %>%
  huxtable::print_md(max_width = 80)
```

Table: (\#tab:iatab) Genotypic diversity and Linkage Disequilibrium summary for geographic populations arranged by abundance after clone-correction by a hierarchy of Region/Source/Host/Year. 
Pop = Population, N = number of individuals (number of MLH in parentheses), 
eMLH = expected number of MLHs based on rarefaction at 10 individuals (standard error in parentheses), 
H = Shannon-Weiner Index, G = Stoddardt and Taylor's Index, 
$\lambda$ = Simpson's Index, 
*h* = Nei's 1978 gene diversity, 1
$E_5$ = Evenness,
$\bar{r}_d$ = standardized index of association.
An asterix indicates a significant value of $\bar{r}_d$ after 999 permutations, $P \leq 0.001$.
